packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -129,24 +145,19 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
      * - gcData: The garbage collection data of this node that is required for running GC.
      */
     private async loadInitialGCSummaryDetails() {
-        // If referenceUsedRoutes exists, don't do anything because we have already initialized.
-        if (this.referenceUsedRoutes !== undefined) {
+        if (this.gcDetailsInInitialSummaryP === undefined) {
```

Comment:
Changed this because it was called multiple times if the initial GC details did not have used routes. This should be called only the first time GC is run.